src/Sarif/Core/PropertyBagHolder.cs

```diff
@@ -99,7 +99,9 @@ public T GetProperty<T>(string propertyName)
                         propertyName));
             }
 
-            return JsonConvert.DeserializeObject<T>(Properties[propertyName].SerializedValue);
+            SerializedPropertyInfo propValue = Properties[propertyName];
+
+            return propValue == null ? default : JsonConvert.DeserializeObject<T>(propValue.SerializedValue);
```

Comment:
>default [](start = 39, length = 7)

This makes sense for reference types but not for object types. Given `{ "properties": { "a": null } }`, I expect `GetProperty<int>("a")` to throw rather than returning 0.

I suggest we fix this (and add another test) by modifying line 104 as follows:
```
if (propValue == null)
{
    if (typeof<T>.IsValueType)
    {
        throw new InvalidOperationException(
            string.Format(
                CultureInfo.CurrentCulture,
                SdkResources.PropertyOfValueTypeCannotBeNull,
                propertyName,
                typeof(T).FullName));
    }

    return null;
}
else
{
    return JsonConvert.DeserializeObject<T>(propValue.SerializedValue)
}
```

No doubt this could be shortened with `??` and or `? : ` and or a `throw` expression, but it's complicated enough that I think it's better spelled out in full.

Message string: "The property bag contains a property {0} with the value null. It cannot be converted to the value type {1}."

Reply:
Adjusted per recommendations and added a test for value types